### ISSUE TITLE
Unknown AnalysisClassHierarchy.Untracked(_) exception occurs at analysis

### ISSUE THREAD

**Issue by angelinag** [facebook/pyre-check#378]
Hello, urgently trying to find a fix for an unidentified and undocumented error.
Running pysa analysis and getting this error:

```Uncaught exception: ("AnalysisClassHierarchy.Untracked(_)")
ƛ Uncaught exception:
ƛ 
ƛ   AnalysisClassHierarchy.Untracked(_)
ƛ 
ƛ Raised at file "string.ml", line 115, characters 19-34
ƛ Called from file "src/sexp.ml", line 113, characters 13-47
ƛ Client exited with error code 1```
```

pyre config file:
```
{
  "source_directories": [
    "."
  ],
  "taint_models_path": ".",
  "exclude": "/home/zeus/work/test-protecc/env/lib/python3.9/.*"
}
```

Our model:
```
model_content = """
protecc.vortex.Vortex.SECRETS: TaintSource[Secret]
protecc.vortex.Vortex.endpoints: TaintSink[Endpoint]
"""
```

The class its modelling:
```
...
class Vortex:
    def __init__(self):
        self.SECRETS = {}   # Populate with sources
        self.endpoints = set() # Populate with sinks
...
```

We tried tweaking a lot of things and we keep getting the same error.
Only found 3 mentions of this problem so far [here](https://github.com/facebook/pyre-check/issues/344), [here at the end of the page](https://github.com/zulip/zulip/pull/16363) and also [here](https://github.com/facebook/pyre-check/issues/350). None have been useful so far.

**Comment by gbleaney** [comment]
@angelinag we're going to do some work to make that error message a lot more helpful. Thanks for pointing this out.

In the mean time, your attempt at excluding your venv in via `taint.config` seems like the right approach, because odds are it's something in the venv that's causing the exception. I wonder if the exclude isn't working though. To test that, can you try two different approaches:

1. Update the exclude to something like `.*/env/.*`
2. Just remove the venv from the project you're analyzing entirely

If you're still getting errors after trying both of those, perhaps the problematic code really is present in your project. If that's the case, can you try doing a sort of binary search for the bug by deleting one half the code and then the other, and seeing if it makes the error go away? If you find the problematic file/lines, we can use that as a test case to get a fix out for you.

**Comment by SaurusXI** [comment]
@gbleaney I'm a part of this project also. I've got an update on this - turns out the problem was within our own library, specifically a misplaced import statement.
As you've already mentioned, the issue here is that there is no meaningful exception that can help with debugging. I might have a helpful suggestion for this - I was able to identify what was wrong by running a `pyre` server in the project directory. That gave a much more meaningful `unable to find module ...`  exception.
I'm not sure if this can be generalized to dealing with all `AnalysisClassHierarchy.Untracked(_)` exceptions, or even if it is the most helpful way of doing this, but as far as I know `pyre analyze` also runs the `pyre` type checker, so it might be possible to propagate exceptions from the type checker?
